src/Psalm/Internal/Analyzer/Statements/Expression/ArrayAnalyzer.php

```diff
@@ -127,7 +106,7 @@ public static function analyze(
             $item_key_type = null;
         }
 
-        if ($array_creation_info->item_value_atomic_types) {
+        if (!empty($array_creation_info->item_value_atomic_types)) {
```

Comment:
Could we use anything but empty? I very much dislike this function (it make sense sometimes, but rarely)

Reply:
Oh, yeah, bad habit because I'm used to similar functions specifically for arrays/vecs/lists in other languages, but I'm also not a fan of implicit conversion of array to bool either, I'll try to remember to do `count(...) === 0` in the future.

Reply:
Looks like there are a lot of `empty` usages around that could probably be changed, would it be worthwhile to add some sort of `bannedFunctions` Psalm config, or do other tools handle that well enough that we don't want to? For stuff like this you don't want to ban it entirely, you just want a warning that you can suppress, so disabling the function in PHP isn't possible.

Reply:
I removed some a while ago when it seemed safe: https://github.com/vimeo/psalm/pull/4211/files

There's indeed a few more that popped up since then. I actually have a Psalm plugin for getting rid of those somewhere !

I guess we could add it to forbidden functions yeah: https://psalm.dev/docs/running_psalm/issues/ForbiddenCode/

Reply:
Aha! I _thought_ we had something like that. Looks like that's not documented on the configuration page, so that's why I missed it.